tmt/steps/discover/sources.py

```diff
@@ -0,0 +1,102 @@
+import os
+import re
+import shutil
+
+import click
+import fmf
+
+import tmt
+import tmt.beakerlib
+import tmt.steps.discover
+
+from tmt.steps.discover import DiscoverPlugin
+
+class DiscoverSources(DiscoverPlugin):
+    """
+    Get sources extracted from dist-git and delegate discovery step
+
+        discover:
+            how: sources
+            delegate:
+                how: fmf
+
+    Full config example:
+
+        discover:
+            how: sources
+            delegate:
+                how: fmf
+            shell: |
+                cd ./*
+                fmf init
+            url: https://src.fedoraproject.org/rpms/tmt.git
+    """
+
+    # Supported methods
+    _methods = [tmt.steps.Method(name='sources', doc=__doc__, order=50)]
+
+    def go(self):
+        """ Prepare sources """
+        super(DiscoverSources, self).go()
+
+        sourcedir = os.path.join(self.workdir, 'TEMPORARY')
+        rpmbuilddir = os.path.join(self.workdir, 'temp_tests')
+        testdir = os.path.join(self.workdir, 'tests')
+
+        url = self.get('url')
+
+        if url:
+            self.debug(f"Clone '{url}' to '{sourcedir}'.")
+            self.run(
+                ['git', 'clone', url, sourcedir],
+                shell=False, env={"GIT_ASKPASS": "echo"})
+        else:
+            # Copy current directory
+            self.debug(f"Copy '.' to '{sourcedir}'.")
+            shutil.copytree('.', sourcedir)
+        # Download sources
+        self.run(
+            ['fedpkg', 'sources', '--outdir', sourcedir],
+            cwd=sourcedir if url else '.',
+            shell=False
+        )
+
+        # Run rpmbuild
+        self.run([
+            "rpmbuild",
+            "-bp",
+            "--define",
+            f"_builddir {rpmbuilddir}",
+            "--define",
+            f"_sourcedir {sourcedir}",
+            "--define",
+            f"_specdir {sourcedir}",
+            "tmt.spec"],
+            cwd=sourcedir,
+            shell=False)
+
+        # Move extracted sources
+        shutil.move(
+            os.path.join(
+                rpmbuilddir,
+                os.listdir(rpmbuilddir)[0]
+            ),
+            testdir
+        )
+
+        # Run shell if it was there
+        shell = self.get('shell')
+        if shell:
+            self.run(shell, cwd=testdir)
```

Comment:
So basically, you run any abitrary command on host.

This is what I wanted to avoid.

Reply:
It is not required - just a hotfix because tmt sources doesn't contain .fmf/version.

Reply:
And frankly  - we need `rpmbuild -bp` to run on host. If you really want to put some malicious code there you can / and will be able to.
This discover plugin is optional - you don't need to install it

Reply:
You're right, I think it should use some method of isolation then.

Reply:
What if it runs in mock?